File: wfe-restapi/src/main/java/ru/runa/wfe/rest/dto/WfDefinitionDto.java
package ru.runa.wfe.rest.dto;

import java.util.Date;
import lombok.Data;

@Data
public class WfDefinitionDto {
    private Long id;
    private Long versionId;
    private String name;
    private String description;
    private String[] categories;
    private Long version;
    private boolean hasHtmlDescription;
    private boolean hasStartImage;
    private boolean hasDisabledImage;
    private boolean subprocessOnly;
    private boolean canBeStarted;
    private Date createDate;
    private WfUserDto createActor;
    private Date updateDate;
    private WfUserDto updateActor;
    private Date subprocessBindingDate;
    private Integer secondsBeforeArchiving;
}

File: wfe-spa/src/ts/WfDefinitionDto.ts
import { ExecutorDto } from './ExecutorDto';

export class WfDefinitionDto {
    id: number|null = null;
    versionId: number|null = null;
    name: string = '';
    description: string = '';
    categories: Array<string>|null = '';
    version: number|null = null;
    canBeStarted: boolean = false;
    createDate: Date|null = null;
    createActor: ExecutorDto|null = null;
    updateDate: Date|null = null;
    updateActor: ExecutorDto|null = null;

    getInfo(): object {
        return {
            'ID': this.versionId,
            'Имя': this.name,
            'Описание': this.description,
            'Создана': this.createDate ? new Date(this.createDate).toLocaleString() : '',
            'Автор создания': this.createActor ? this.createActor.fullName : '',
            'Обновлена': this.updateDate ? new Date(this.updateDate).toLocaleString() : '',
            'Автор обновления': this.updateActor ? this.updateActor.fullName : '',
        };
    }
}